curriculum/challenges/english/08-coding-interview-prep/rosetta-code/zhang-suen-thinning-algorithm.english.md

```diff
@@ -44,49 +44,51 @@ It produces the thinned output:
-<table border="1">
-  <tr><td>P9</td><td>P2</td><td>P3</td></tr>
-  <tr><td>P8</td><td><b>P1</b></td><td>P4</td></tr>
-  <tr><td>P7</td><td>P6</td><td>P5</td></tr>
-</table>
-Obviously the boundary pixels of the image cannot have the full eight neighbours.
 
-    Define $A(P1)$ = the number of transitions from white to black, (0 -> 1) in the sequence P2,P3,P4,P5,P6,P7,P8,P9,P2. (Note the extra P2 at the end - it is circular).
+| P9 | P2 | P3 |
+|:--:|:--:|:--:|
+| <b>P8</b> | <b>P1</b> | <b>P4</b> |
+| <b>P7</b> | <b>P6</b> | <b>P5</b> |
 
-
-    Define $B(P1)$ = the number of black pixel neighbours of P1. ( = sum(P2 .. P9) )
+Obviously the boundary pixels of the image cannot have the full eight neighbours.
+  <ul>
+    <li>Define $A(P1)$ = the number of transitions from white to black, (0 -> 1) in the sequence P2,P3,P4,P5,P6,P7,P8,P9,P2. (Note the extra P2 at the end - it is circular).</li>
+    <li>Define $B(P1)$ = the number of black pixel neighbours of P1. ( = sum(P2 .. P9) )</li>
+  </ul>
 
 <h3>Step 1:</h3>
 All pixels are tested and pixels satisfying all the following conditions (simultaneously) are just noted at this stage.
-  (0) The pixel is black and has eight neighbours
-  (1) $2 <= B(P1) <= 6$
-  (2) $A(P1) = 1$
-  (3) At least one of P2 and P4 and P6 is white
-  (4) At least one of P4 and P6 and P8 is white
+  <ol>
+    <li>The pixel is black and has eight neighbours</li>
+    <li>$2 <= B(P1) <= 6$</li>
+    <li>$A(P1) = 1$</li>
+    <li>At least one of P2 and P4 and P6 is white</li>
+    <li>At least one of P4 and P6 and P8 is white</li>
+  </ol>
 After iterating over the image and collecting all the pixels satisfying all step 1 conditions, all these condition satisfying pixels are set to white.
+
 <h3>Step 2:</h3>
 All pixels are again tested and pixels satisfying all the following conditions are just noted at this stage.
-  (0) The pixel is black and has eight neighbours
-  (1) $2 <= B(P1) <= 6$
-  (2) $A(P1) = 1$
-  (3) At least one of P2 and P4 and '''P8''' is white
-  (4) At least one of '''P2''' and P6 and P8 is white
+  <ol>
+    <li>The pixel is black and has eight neighbours</li>
+    <li>$2 <= B(P1) <= 6$</li>
+    <li>$A(P1) = 1$</li>
+    <li>At least one of P2 and P4 and '''P8''' is white</li>
+    <li>At least one of '''P2''' and P6 and P8 is white</li>
+  </ol>
 After iterating over the image and collecting all the pixels satisfying all step 2 conditions, all these condition satisfying pixels are again set to white.
 Iteration:
 If any pixels were set in this round of either step 1 or step 2 then all steps are repeated until no image pixels are so changed.
-<p>
-Task:
-Write a routine to perform Zhang-Suen thinning on an image matrix of ones and zeroes.
-</p>
 </section>
 
 ## Instructions
 <section id='instructions'>
-
+Write a routine to perform Zhang-Suen thinning on an image matrix of ones and zeroes.
```

Comment:
The image used for the tests doesn't seem to use ones and zeros, so the text here should be changed to something like, "Write a routine to perform Zhang-Suen thinning on the provided image matrix."